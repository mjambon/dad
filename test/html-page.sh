./vanity -s \
  --title='Glossary' \
  --include-in-header=in-header.html \
  --include-before-body=before-body.html \
  --include-after-body=after-body.html \
  < test.yml
