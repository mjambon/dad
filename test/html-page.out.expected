<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Glossary</title>
<!--
   This is meant to be included in the <head> section of an HTML document.
-->

<meta name="viewport"
      content="width=device-width, initial-scale=1.0, user-scalable=yes">

<style>
/* Use fixed-width body for readability. */
body {
  margin: 6.5em auto 2.5em auto;
  padding: 0.5em 1.5em;

  max-width: 70ex;
  text-align: left;
  font-family: 'Droid Serif', serif;

  /* Set the background to something different the the default github
     background color (white) so that a screenshot is identified as
     such. */
  background-color: #fefaff;
  color: black;
}

h1, h2, h3, h4 {
  margin: 1.5em 0em 0.8em 0em;
  line-height: 1.1;
}

/* Make page readable on mobile devices. */
@media only screen and (max-width: 480px) {
  body {
    margin: 0px;
  }
}

/* Fit the graph nicely */
img {
  height: auto;
  max-width: 100%;
  display: block;
  margin: 0em auto 0em;
}

/*
   The various HTML elements created by 'vanity' have their own class
   name, allowing them to be styled.
*/
a.vanity-term-link,
a.vanity-term-link:visited {
  color: black;
  font-weight: bold;
}

a.vanity-term-link:hover,
a.vanity-term-link:active {
  color: #f2a;
}
</style>
</head>
<body>
<h1>Glossary</h1><p>
This is a sample glossary rendered with
<a href="https://github.com/mjambon/vanity">vanity</a>, a program meant to
encourage the use of acyclic definitions. All terms rendered as links
are guaranteed to have been defined earlier.
</p>


<p class="ad-def">
  <a name="ad-6675726e6974757265"></a><strong class="ad-term">furniture</strong>:
  <span class="ad-contents">collective noun designating all sorts of large but movable items used in indoor settings with a primary non-decorative function.
</span>
</p>

<p class="ad-def">
  <a name="ad-7461626c65"></a><strong class="ad-term">table</strong>:
  <span class="ad-contents">a <a href="#ad-6675726e6974757265" class="ad-term-link">piece of furniture</a> with a horizontal surface lying above the floor that was designed to support smaller objects.
</span>
</p>

<p class="ad-def">
  <a name="ad-73656174"></a><strong class="ad-term">seat</strong>:
  <span class="ad-contents">a <a href="#ad-6675726e6974757265" class="ad-term-link">piece of furniture</a> designed for a single person to sit on it.
</span>
</p>

<p class="ad-def">
  <a name="ad-6368616972"></a><strong class="ad-term">chair</strong>:
  <span class="ad-contents">a mobile <a href="#ad-73656174" class="ad-term-link">seat</a> with a backrest.
</span>
</p>

<p class="ad-def">
  <a name="ad-73746f6f6c"></a><strong class="ad-term">stool</strong>:
  <span class="ad-contents">a <a href="#ad-73656174" class="ad-term-link">seat</a> without a backrest.
</span>
</p>

<p class="ad-def">
  <a name="ad-62656e6368"></a><strong class="ad-term">bench</strong>:
  <span class="ad-contents">an item designed for several people to sit on, with a hard surface.
</span>
</p>

<p class="ad-def">
  <a name="ad-636f756368"></a><strong class="ad-term">couch</strong>:
  <span class="ad-contents">a <a href="#ad-6675726e6974757265" class="ad-term-link">piece of furniture</a> designed for multiple people to sit on, with a cushy surface.
</span>
</p>

<p>
  Here's the directed acyclic graph (DAG) representing all the terms:
</p>
<p>
  <a href="complex.png"><img src="complex.png"></a>
</p>
</body>
